R/plot.r

```diff
@@ -36,43 +39,40 @@
 #'   evaluation.
 #' @export
 #' @examples
-#' # Generate some sample data, then compute mean and standard deviation
-#' # in each group
+#' # Create a data frame with some sample data, then create a data frame
+#' # containing the mean for each group in the sample data.
 #' set.seed(1)
-#' df <- data.frame(
-#'   gp = factor(rep(letters[1:3], each = 10)),
-#'   y = rnorm(30)
+#' sample_df <- data.frame(
+#'   group = factor(rep(letters[1:3], each = 10)),
+#'   value = rnorm(30)
 #' )
-#' ds <- do.call(rbind, lapply(split(df, df$gp), function(d) {
-#'   data.frame(mean = mean(d$y), sd = sd(d$y), gp = d$gp)
+#' group_means_df <- do.call(rbind, lapply(split(sample_df, sample_df$group), function(d) {
+#'   data.frame(group_mean = mean(d$value), group = d$group)
 #' }))
-#'
-#' # The summary data frame ds is used to plot larger red points on top
-#' # of the raw data. Note that we don't need to supply `data` or `mapping`
-#' # in each layer because the defaults from ggplot() are used.
-#' ggplot(df, aes(gp, y)) +
+#' 
+#' # Pattern 1
+#' # The group means data frame is used to plot larger red points on top
+#' # of the sample data. Note that we don't need to supply `data =` or `mapping =`
+#' # in each layer because the arguments are passed into ggplot() in the default
+#' # positions.
+#' ggplot(sample_df, aes(x = group, y = value)) +
 #'   geom_point() +
-#'   geom_point(data = ds, aes(y = mean), colour = 'red', size = 3)
-#'
+#'   geom_point(group_means_df, aes(y = group_mean), colour = 'red', size = 3)
```

Comment:
This example currently doesn't work. The first argument is the mapping, the second one is the data.
Probably best to be explicit and give `data = group_means_df`. My suggestion:
```r
  geom_point(
    aes(y = group_mean),
    data = group_means_df, 
    colour = 'red', size = 3
  )
```

Reply:
My bad. I was confused by the comment:
```Note that we don't need to supply `data` or `mapping` in each layer because the defaults from ggplot() are used.```, and further confused by the ordering of the `data` and `mapping` arguments being reversed in the `ggplot()` and `geom_point()` functions. And I should have tested my changes!

Does this comment work:
```Note that we don't need to supply `mapping` or `data` arguments in the first geom_point() layer because those arguments are passed along from the arguments supplied in the ggplot() call.```

Also, how do you feel about having explicit `mapping =` keywords in the appropriate places for the `ggplot()` and `geom_point()` calls? I know both `mapping =` and `data =` are often left out in practice, but as a person learning these patterns for the first time (who I think are the most likely readers of this documentation), it might be helpful to have them in there. (At least, they're helpful to me.)

Reply:
> Also, how do you feel about having explicit mapping = keywords

Yeah I think that is a good idea. Especially when it is the first time the argument is shown in a new position.

> To help clarify, does this comment work?

I think the word 'arguments' is perhaps used 1 too many times, but I like the clarification in general.
Perhaps something like this?

```
Note that we don't need to supply `mapping` or `data` arguments in the first geom_point() layer
 because those arguments are passed along from the main `ggplot()` call`
```